### ISSUE TITLE
[Question] Support `?` in sql

### ISSUE THREAD

**Issue by haunt98** [sqlfluff/sqlfluff#4015]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

Can sqlfluff support `?` in sql query like https://go.dev/doc/database/querying?

For example: 

```go
rows, err := db.Query("SELECT * FROM album WHERE artist = ?", artist)
if err != nil {
    return nil, err
}
defer rows.Close()
```

Currently sqlfluff failed to parse

<img width="739" alt="image" src="https://user-images.githubusercontent.com/5424272/199381985-96045f6d-0efb-4d49-ad10-b85898b00187.png">

### Dialect

MySQL

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by barrywhart** [comment]
I don't think this change should be necessary. You should be able to use the `placeholder` templater with `param_style = question_mark`.

https://docs.sqlfluff.com/en/stable/configuration.html#placeholder-templating

**Comment by haunt98** [comment]
Thanks @barrywhart, let me try again!

**Comment by WittierDinosaur** [comment]
Closing as placeholder templater should owrk, and has been awaiting feedback for several months

**Comment by haunt98** [comment]
Sorry @WittierDinosaur I forgot about this. It's worked as I expected, thank you!
